lighthouse-core/audits/byte-efficiency/unused-javascript.js

```diff
@@ -0,0 +1,184 @@
+      );
+    } else {
+      let endIndex = 1;
+      while (endIndex < ranges.length && ranges[endIndex].startOffset < firstRange.endOffset) {
+        endIndex++;
+      }
+
+      const spanningRanges = ranges.slice(1, endIndex);
+      const restOfRanges = ranges.slice(endIndex);
+      if (firstRange.isUsed) {
+        // Nested ranges may potentially be unused which we need to account for.
+        const earliestStart = secondRange.startOffset;
+        const latestEnd = spanningRanges.reduce((max, next) => {
+          return max.endOffset >= next.endOffset ? max : next;
+        }).endOffset;
+        const nextStart = (restOfRanges[0] && restOfRanges[0].startOffset) || firstRange.endOffset;
+
+        const leftEdge = {used: earliestStart - firstRange.startOffset, unused: 0};
+        const rightEdge = {used: nextStart - latestEnd, unused: 0};
+        const inner = this.findUsedAndUnused(spanningRanges);
+        const rest = this.findUsedAndUnused(restOfRanges);
+        return this._add(this._add(leftEdge, rightEdge), this._add(inner, rest));
+      } else {
+        // Nested ranges won't be used if the entire parent wasn't used, just skip the children.
+        return this._add(firstRange, this.findUsedAndUnused(restOfRanges));
+      }
+    }
+  }
+
+  /**
+   * @param {!Object} script
+   * @param {{transferSize: number}} networkRecord
+   */
+  static computeWaste(script, networkRecord) {
+    const url = URL.getDisplayName(script.url);
+
+    let numUnusedFunctions = 0;
+    const ranges = [];
+    script.functions.forEach(func => {
+      let functionIsUsed = false;
+      func.ranges.forEach(range => {
+        const isUsed = range.count > 0;
+        const span = range.endOffset - range.startOffset;
+        const used = isUsed ? span : 0;
+        const unused = isUsed ? 0 : span;
+
+        functionIsUsed = functionIsUsed || isUsed;
+        ranges.push(Object.assign({isUsed, span, used, unused}, range));
+      });
+
+      if (!functionIsUsed) {
+        numUnusedFunctions++;
+      }
+    });
+
+    ranges.sort((rangeA, rangeB) => {
+      return rangeA.startOffset - rangeB.startOffset || rangeB.endOffset - rangeA.endOffset;
+    });
+
+    const results = this.findUsedAndUnused(ranges);
+    const wastedRatio = (results.unused / (results.unused + results.used)) || 0;
+    const totalBytes = networkRecord.transferSize;
+    const wastedBytes = Math.round(totalBytes * wastedRatio);
+
+    return {
+      url,
+      totalBytes,
+      wastedBytes,
+      wastedPercent: 100 * wastedRatio,
+      numUnused: numUnusedFunctions,
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {{results: !Array<!Object>, tableHeadings: !Object}}
+   */
+  static audit_(artifacts, networkRecords) {
+    const resultsMap = artifacts.JsUsage.reduce((results, script) => {
+      const networkRecord = networkRecords.find(record => record.url === script.url);
+      if (!networkRecord) {
+        return results;
```

Comment:
how often does this happen? Our injected scripts, extension scripts...what causes this?